src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,333 @@
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(ReadOnlyMemory<char> memory, out Single value)
+        {
+            var res = DoubleParser.Parse(out value, memory.Span);
+            Contracts.Assert(res != DoubleParser.Result.Empty || value == 0);
+            return res <= DoubleParser.Result.Empty;
+        }
+
+        /// <summary>
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(ReadOnlyMemory<char> memory, out Double value)
+        {
+            var res = DoubleParser.Parse(out value, memory.Span);
+            Contracts.Assert(res != DoubleParser.Result.Empty || value == 0);
+            return res <= DoubleParser.Result.Empty;
+        }
+
+        public static uint Hash(ReadOnlySpan<char> span, uint seed) => Hashing.MurmurHash(seed, span);
+
+        public static NormStr AddToPool(ReadOnlyMemory<char> memory, NormStr.Pool pool)
+        {
+            Contracts.CheckValue(pool, nameof(pool));
+            return pool.Add(memory);
+        }
+
+        public static NormStr FindInPool(ReadOnlyMemory<char> memory, NormStr.Pool pool)
+        {
+            Contracts.CheckValue(pool, nameof(pool));
+            return pool.Get(memory);
+        }
+
+        public static void AddToStringBuilder(ReadOnlyMemory<char> memory, StringBuilder sb)
+        {
+            Contracts.CheckValue(sb, nameof(sb));
+            if (!memory.IsEmpty)
+                sb.AppendAll(memory);
+        }
+
+        public static void AddLowerCaseToStringBuilder(ReadOnlyMemory<char> memory, StringBuilder sb)
+        {
+            Contracts.CheckValue(sb, nameof(sb));
+
+            if (!memory.IsEmpty)
+            {
+                int min = 0;
+                int j;
+                var span = memory.Span;
+                for (j = min; j < memory.Length; j++)
+                {
+                    char ch = CharUtils.ToLowerInvariant(span[j]);
+                    if (ch != span[j])
+                    {
+                        sb.Append(memory, min, j - min).Append(ch);
+                        min = j + 1;
+                    }
+                }
+
+                Contracts.Assert(j == memory.Length);
+                if (min != j)
+                    sb.Append(memory, min, j - min);
+            }
+        }
+
+        private static bool ContainsChar(char ch, char[] rgch)
+        {
+            Contracts.AssertNonEmpty(rgch, nameof(rgch));
+
+            for (int i = 0; i < rgch.Length; i++)
+            {
+                if (rgch[i] == ch)
+                    return true;
+            }
+            return false;
+        }
+
+        public static StringBuilder AppendAll(this StringBuilder sb, ReadOnlyMemory<char> memory) => Append(sb, memory, 0, memory.Length);
+
+        public static StringBuilder Append(this StringBuilder sb, ReadOnlyMemory<char> memory, int startIndex, int length)
```

Comment:
>ReadOnlyMemory [](start = 66, length = 14)

Also, you should just take a `ReadOnlySpan<char>` here. That way if the caller only has a span, they can still call this method.